### PR TITLE
Log request and response headers

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Item 2 of #4.

cc @cesarbs @halter73

### CODE REVIEW

File: src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs [hunk 1]
```diff
@@ -655,6 +708,12 @@ private Task WriteSuffix()
 
             end.CopyFrom(_httpVersion == HttpVersionType.Http1_1 ? _bytesHttpVersion1_1 : _bytesHttpVersion1_0);
             end.CopyFrom(statusBytes);
+
+            if (Log.IsEnabled(LogLevel.Debug))
+            {
+                Log.LogDebug(ResponseToString());
```
Comment (JunTaoLuo) [inline src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs]: Possibly need eventids or something more than just a string

**Comment by JunTaoLuo** [comment]
I thought it was best to log the headers and such closer to when they are actually read and written but I suppose when the HttpContext was created/disposed is good enough. I'll close this PR and open a new one in Hosting and remove the logging of headers in WebListener.

**Comment by halter73** [comment]
@davidfowl I think it would be a good idea to expose a connection and request ID via the IHttpConnectionFeature and IHttpRequestFeature respectively. This should make it easier to correlate logs from different layers. I can open issues in Abstractions and Kestrel if you agree.

**Comment by davidfowl** [comment]
`IHttpConnectionFeature.ConnectionId` I agree with but I don't think there should be a request id on the `IHttpRequestFeature`. I prefer what we have right now as it doesn't need to be implemented in the sever itself and it's not *required* for anything to function.

**Comment by Tratcher** [comment]
I expect we'll need ConnectionId at some point. We already have IHttpRequestIdentifierFeature for request id. I agree that it would make correlation easier if the Kestrel implemented IHttpRequestIdentifierFeature. WebListener does.

**Pull request closed without merging**